pkgs/by-name/nd/ndstrim/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, fetchpatch
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ndstrim";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "Nemris";
+    repo = "ndstrim";
+    rev = "v${version}";
+    hash = "sha256-KgtejBbFg6+klc8OpCs1CIb+7uVPCtP0/EM671vxauk=";
+  };
+
+  patches = [
+    # https://github.com/Nemris/ndstrim/pull/1
+    (fetchpatch {
+      name = "update-cargo-lock.patch";
+      url = "https://github.com/Nemris/ndstrim/commit/8147bb31a8fb5765f33562957a61cb6ddbe65513.patch";
+      hash = "sha256-HsCc5un9dg0gRkRjwxtjms0cugqWhcTthGfcF50EgYA=";
+    })
+  ];
+
+  # TODO: remove once the patch above is merged
+  postPatch = ''
+    cp /build/source/Cargo.lock /build/ndstrim-${version}-vendor.tar.gz/Cargo.lock
```

Comment:
I never seen such usage of a hardcoded reference to the build directory...

Reply:
Not sure why this issue is happening either but it was the only way I found to fix it.

Reply:
Have you tried to vendor this patch, instead of fetching it?

Reply:
I added more details, and also used a trick that can be found in other packages in `nixpkgs`.

I think this should probably be more acceptable, WDYT @AndersonTorres.

Reply:
> Have you tried to vendor this patch, instead of fetching it?

This is not an issue upstream. I think our logic that does vendor in nixpkgs is buggy and it is applying before the `patchPhase`.

Reply:
> I think this should probably be more acceptable, WDYT

Looks fine to me.

> vendoring dependencies before the patchPhase

Running a raw `patch -p1` command at `postUnpack` is too ugly...

Maybe this issue deserves to be opened to the Nix Rust maintainers?